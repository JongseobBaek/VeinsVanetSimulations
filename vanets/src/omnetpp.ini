[General]
cmdenv-express-mode = true
cmdenv-autoflush = true
cmdenv-status-frequency = 10000000s

#tkenv-default-config = debug
#tkenv-default-run = 1
tkenv-image-path = bitmaps
ned-path = .

network = RSUScenario

##########################################################
#            Simulation parameters                       #
########################mac##################################
debug-on-errors = true
print-undisposed = false
record-eventlog = true

#6000s
sim-time-limit = 200s

#Extra parameters for testing signals
check-signals=true

#Testing parameters
**.scalar-recording = true
**.vector-recording = true
**.throughputSignal.scalar-recording = true

**.debug = true
**.coreDebug = false

*.playgroundSizeX = 2500m
*.playgroundSizeY = 2500m
*.playgroundSizeZ = 50m

##########################################################
# Annotation parameters                                  #
##########################################################
*.annotations.draw = false

##########################################################
# Obstacle parameters                                    #
##########################################################
*.obstacles.debug = false

##########################################################
#WorldUtilityParameters									 #
##########################################################
*.world.useTorus = false
*.world.use2D = false

##########################################################
#TraCIScenarioManager Parameters						 #
##########################################################
*.manager.updateInterval = 0.1s
*.manager.host = "localhost"
*.manager.port = 9999
*.manager.moduleType = "vanets.src.CarWithMetrics"
*.manager.moduleName = "node"
*.manager.moduleDisplayString = ""
*.manager.autoShutdown = true
*.manager.margin = 25
*.manager.numVehicles = 5
*.manager.launchConfig = xmldoc("erlangen.launchd.xml")

##########################################################
# RSU SETTINGS											 #
##########################################################
*.rsu[0].mobility.x = 2000
*.rsu[0].mobility.y = 2000
*.rsu[0].mobility.z = 3
*.rsu[*].applType = "TraCIDemoRSU11p"
*.rsu[*].appl.debug = false
*.rsu[*].appl.headerLength = 256 bit
*.rsu[*].appl.sendBeacons = false
*.rsu[*].appl.dataOnSch = false
*.rsu[*].appl.sendData = true
*.rsu[*].appl.beaconInterval = 1s
*.rsu[*].appl.beaconPriority = 3
*.rsu[*].appl.dataPriority = 2
*.rsu[*].appl.maxOffset = 0.005s

*.metricsApplication.mobility.x = 2000
*.metricsApplication.mobility.y = 2000
*.metricsApplication.mobility.z = 3
*.metricsApplication.applType = "TraCIDemoRSU11p"
*.metricsApplication.appl.debug = false
*.metricsApplication.appl.headerLength = 256 bit
*.metricsApplication.appl.sendBeacons = false
*.metricsApplication.appl.dataOnSch = false
*.metricsApplication.appl.sendData = true
*.metricsApplication.appl.beaconInterval = 1s
*.metricsApplication.appl.beaconPriority = 3
*.metricsApplication.appl.dataPriority = 2
*.metricsApplication.appl.maxOffset = 0.005s


##########################################################
#            11p specific parameters                     #
#                                                        #
#                    NIC-Settings                        #
##########################################################
*.connectionManager.pMax = 20mW
*.connectionManager.sat = -89dBm
*.connectionManager.alpha = 2.0
*.connectionManager.carrierFrequency = 5.890e9 Hz
*.connectionManager.sendDirect = true

*.**.nic.mac1609_4.useServiceChannel = false
*.**.nic.mac1609_4.scalar-recording = true
 
*.**.nic.mac1609_4.txPower = 20mW
*.**.nic.mac1609_4.bitrate = 6Mbps
*.**.nic.phy80211p.sensitivity = -89dBm
*.**.nic.phy80211p.maxTXPower = 100mW
*.**.nic.mac1609_4.queueSize = ${1,2,5}
*.**.nic.mac1609_4.vector-recording = true

*.**.nic.phy80211p.useThermalNoise = true
*.**.nic.phy80211p.thermalNoise = -110dBm
*.**.nic.phy80211p.decider = xmldoc("config.xml")
*.**.nic.phy80211p.analogueModels = xmldoc("config.xml")
*.**.nic.phy80211p.usePropagationDelay = true

##########################################################
##########################################################
*.node[*].applType = "TraCIDemo11p"
*.node[*].appl.debug = false
*.node[*].appl.headerLength = 256 bit
*.node[*].appl.sendBeacons = false
*.node[*].appl.dataOnSch = false
*.node[*].appl.sendData = true
*.node[*].appl.beaconInterval = 1s
*.node[*].appl.beaconPriority = 3
*.node[*].appl.dataPriority = 2
*.node[*].appl.maxOffset = 0.005s



##########################################################
#                      Mobility                          #
##########################################################
*.node[*].veinsmobilityType = "org.car2x.veins.modules.mobility.traci.TraCIMobility"
*.node[*].mobilityType = "TraCIMobility"
*.node[*].mobilityType.debug = true
*.node[*].veinsmobilityType.debug = true
*.node[*].veinsmobility.x = 0
*.node[*].veinsmobility.y = 0
*.node[*].veinsmobility.z = 1.895
*.node[*0].veinsmobility.accidentCount = 1
*.node[*0].veinsmobility.accidentStart = 10s #75s
*.node[*0].veinsmobility.accidentDuration = 30s

[Config nodebug]
description = "default settings"
**.debug = false
**.coreDebug = false
*.annotations.draw = false

[Config debug]
description = "(very slow!) draw and print additional debug information"
**.debug = true
**.coreDebug = true
*.annotations.draw = true

[Config B_400Bytes_P_1000Bytes_Q_5]
description = "BeaconLength 400Bytes, ServicePacketLength 1000Bytes, QueueSize 5"
**.debug = false
**.coreDebug = false
*.annotations.draw = false
*.rsu[*].appl.beaconLengthBits = 3200bit
*.node[*].appl.beaconLengthBits = 3200bit
*.rsu[*].appl.dataLengthBits = 8000bit
*.node[*].appl.dataLengthBits = 8000bit
*.rsu[*].appl.dataOnSch = true
*.node[*].appl.dataOnSch = true
*.**.nic.mac1609_4.queueSize = 5
*.**.nic.mac1609_4.useServiceChannel = true
*.manager.launchConfig = xmldoc("singlechannel/LowDensityScenario.launchd.xml")

