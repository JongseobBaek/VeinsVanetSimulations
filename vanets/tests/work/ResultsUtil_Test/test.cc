
#include <omnetpp.h>

#include <ResultsUtilTest.h>



namespace ResultsUtil_Test {



class Test : public cSimpleModule
{
    public:
        Test() : cSimpleModule(32768) {}
        virtual void activity();
};

Define_Module(Test);

void Test::activity()
{

cout<< "WHHAT LKDSJFLSKDJFL:S?"<<endl;
ResultsUtilTest* resultsUtilTest = new ResultsUtilTest();

resultsUtilTest -> testTotalAvgReturnsZeroWhenThroughputPpsIsZero();

resultsUtilTest -> testTotalAvgIsNonzeroWhenOneOfTwoThroughputsIsZero();

resultsUtilTest -> testTotalAvgPrintsErrorWhenTypeNotInFile();

resultsUtilTest -> getTotalAvgFromFileInActualResults();

resultsUtilTest -> getTotalAvgFromLowDensitySingleChSafetyQ5();

resultsUtilTest -> getTotalAvgFromLowDensitySingleChSafetyQ2();

resultsUtilTest -> getTotalAvgFromLowDensitySingleChSafetyQ1();

resultsUtilTest -> getTotalAvgFromLowDensitySingleChSafetyQ100B100Bytes();

//resultsUtilTest -> getTotalAvgFromLowDensitySingleChSafetyQ2B100Bytes();

resultsUtilTest -> getTotalAvgFromLowDensitySingleChSafetyQ1B100Bytes();

resultsUtilTest -> getTotalAvgFromSlidesParamsB400BytesP1000BytesQ5();


}

}; //namespace
