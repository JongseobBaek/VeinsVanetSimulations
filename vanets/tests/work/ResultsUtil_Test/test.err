testTotalAvgPrintsErrorWhenTypeNotInFile	ERROR: Metric type not found in file: randomType:
ERROR: Metric type not found in file: throughputMbps
ERROR: Metric type not found in file: throughputMetricMac
ERROR: Metric type not found in file: throughputMetricMac:last
