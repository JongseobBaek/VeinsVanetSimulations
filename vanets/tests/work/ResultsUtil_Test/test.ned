package vanets.tests.work.ResultsUtil_Test;

simple Test{
	
	@isNetwork(true);
}

