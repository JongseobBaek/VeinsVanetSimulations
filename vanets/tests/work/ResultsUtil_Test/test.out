OMNeT++ Discrete Event Simulation  (C) 1992-2014 Andras Varga, OpenSim Ltd.
Version: 4.6, build: 141202-f785492, edition: Academic Public License -- NOT FOR COMMERCIAL USE
See the license for distribution terms and warranty disclaimer
Setting up Cmdenv...
Loading NED files from .: 2

Preparing for running configuration General, run #0...
Scenario: $repetition=0
Assigned runID=General-0-20160808-14:05:51-8844
Setting up network `Test'...
Initializing...
Initializing module Test, stage 0

Running simulation...
** Event #1  T=0   Test (Test, id=1)
WHHAT LKDSJFLSKDJFL:S?
ZeroThroughput.sca
Opened File: ZeroThroughput.sca
testTotalAvgReturnsZeroWhenThroughputPpsIsZero	throughputTotalAvg: 0
ZeroAndNonzeroThroughput.sca
Opened File: ZeroAndNonzeroThroughput.sca
testTotalAvgIsNonzeroWhenOneOfTwoThroughputsIsZero	throughputTotalAvg: 1.25
TypeNotInFile.sca
Opened File: TypeNotInFile.sca
Getting Rid of warning for AVGRESULT0
Opened File: B_400Bytes_P_1000Bytes_Q_5-0.sca
throughputMbps	TotalAvg: 0.285623
Opened File: B_400Bytes_P_1000Bytes_Q_5-0.sca
throughputMetricMac	TotalAvg: 82.5842
Opened File: B_400Bytes_P_1000Bytes_Q_5-0.sca
throughputMetricMac:last	TotalAvg: 82.5842
throughputMbps TotalAvg: 82.5842
throughputMetricMac TotalAvg: 82.5842
throughputMetricMac:last TotalAvg: 82.5842
Opened File: SingleCH_Safety_B_400Bytes_P_1000Bytes_Q_2-0.sca
throughputMbps TotalAvg: 0.283113
Opened File: SingleCH_Safety_B_400Bytes_P_1000Bytes_Q_2-0.sca
throughputMetricMac TotalAvg: 81.865
Opened File: SingleCH_Safety_B_400Bytes_P_1000Bytes_Q_2-0.sca
throughputMetricMac:last TotalAvg: 81.865
Opened File: SingleCH_Safety_B_400Bytes_P_1000Bytes_Q_1-0.sca
throughputMbps TotalAvg: 0.283113
Opened File: SingleCH_Safety_B_400Bytes_P_1000Bytes_Q_1-0.sca
throughputMetricMac TotalAvg: 81.865
Opened File: SingleCH_Safety_B_400Bytes_P_1000Bytes_Q_1-0.sca
throughputMetricMac:last TotalAvg: 81.865
Opened File: SingleCH_Safety_B_100Bytes_Q_100-0.sca
throughputMbps TotalAvg: 0.0868088
Opened File: SingleCH_Safety_B_100Bytes_Q_100-0.sca
throughputMetricMac TotalAvg: 82.1931
Opened File: SingleCH_Safety_B_100Bytes_Q_100-0.sca
throughputMetricMac:last TotalAvg: 82.1931
Opened File: SingleCH_Safety_B_100Bytes_Q_1-0.sca
throughputMbps TotalAvg: 0
Opened File: SingleCH_Safety_B_100Bytes_Q_1-0.sca
throughputMetricMac TotalAvg: 0
Opened File: SingleCH_Safety_B_100Bytes_Q_1-0.sca
throughputMetricMac:last TotalAvg: 0
Opened File: SlidesParams_B_400Bytes_P_1000Bytes_Q_5-0.sca
throughputMbps TotalAvg: 0.272491
Opened File: SlidesParams_B_400Bytes_P_1000Bytes_Q_5-0.sca
throughputMetricMac TotalAvg: 78.7898
Opened File: SlidesParams_B_400Bytes_P_1000Bytes_Q_5-0.sca
throughputMetricMac:last TotalAvg: 78.7898

<!> No more events -- simulation ended at event #2, t=0.


Calling finish() at end of Run #0...

End.
